internal/direct.go

```diff
@@ -21,50 +22,96 @@ const (
 	authzHeader     = "Authorization"
 	bearer          = "Bearer "
 	gzipFormat      = "gzip"
-	octetStream     = "application/octet-stream"
```

Comment:
Let's still retain these constants and use them as needed rather than use string literals in the constructor.